src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int EMPTY_LIST = 0;
+
+    private Node<T> head;
+
+    private Node<T> tail;
+
+    private int size;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+        this.size = EMPTY_LIST;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == EMPTY_LIST) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= size && index < EMPTY_LIST) {
```

Comment:
The condition in the if statement is incorrect. It should be `if (index < 0 || index > size)`. Currently, it will never throw an `IndexOutOfBoundsException` because the condition cannot be true.